craft_application/grammar.py

```diff
@@ -104,12 +104,24 @@ def process_part(
 
 
 def process_parts(
-    *, parts_yaml_data: dict[str, Any], arch: str, target_arch: str
+    *,
+    parts_yaml_data: dict[str, Any],
+    arch: str,
+    target_arch: str,
+    platform_ids: set[str],
 ) -> dict[str, Any]:
     """Process grammar for parts.
 
-    :param yaml_data: unprocessed snapcraft.yaml.
-    :returns: process snapcraft.yaml.
+    :param yaml_data: The parts data with grammar to process. Grammar is
+        processed in-place in this dictionary.
+    :param arch: The architecture the system is on. This is used as the
+        selector for the 'on' statement.
+    :param target_arch: The architecture the system is to build for. This
+        is the selector for the 'to' statement.
+    :param platform_ids: The identifiers for the current platform to build.
+        These are the selectors for the 'for' statement. Duplicates are ignored.
```

Comment:
minor but imo you don't need to specify that duplicates are ignored when the type of `platform_ids` is a set, since it can't have duplicates to begin with

Reply:
Good catch, I took this from the docstring I wrote in craft-grammar, which takes an iterable.